test/14.errorHandling.js

```diff
@@ -0,0 +1,412 @@
+    const directConnection = await driver.connect({
+      ...config,
+      LBQuery: `round_robin(start_id(${newTE.startId}))`,
+    });
+
+    // kill the TE
+    try {
+      await killTE(TEProcData.pid);
+    } catch (e) {
+      should.not.exist(e);
+    }
+
+    // try to execute a query on the TE
+    let results;
+    try {
+      results = await directConnection.execute(
+        "select getnodeid() from dual"
+      );
+      await results.getRows();
+      await results.close();
+    } catch (e) {
+      err = e;
+      console.error(e);
+    }
+    should.exist(err);
+
+    try {
+      await results.close();
+      await directConnection.close();
+    } catch (e) {
+      // do nothing
+    }
+  });
+
+  it("14.4 Can detect when a TE goes down after query execution", async () => {
+    const newTE = await startTE(JSON.stringify(postData));
+    let err = null;
+    // wait to do anything until the TE is running.
+    let TEProcData;
+    try {
+      TEProcData = await pollForTERunning(newTE.startId)
+    } catch (e) {
+      should.not.exist(e);
+    }
+
+    // create a connection to this TE
+    const directConnection = await driver.connect({
+      ...config,
+      LBQuery: `round_robin(start_id(${newTE.startId}))`,
+    });
+
+    // try to execute a query on the TE, kill before getting results
+    let results;
+    try {
+      results = await directConnection.execute(
+        "select getnodeid() from dual"
+      );
+      await killTE(TEProcData.pid);
+      await results.getRows();
+      await results.close();
+    } catch (e) {
+      err = e;
+      console.error(e);
+    }
+    should.exist(err);
+
+    try {
+      await results.close();
+      await directConnection.close();
+    } catch (e) {
+      // do nothing
+    }
+  });
+
+  it('14.5 Can detect a network timeout on connection', async () => {
+    await netDown();
+    let err;
+    try {
+      const conn = await driver.connect(config);
+      const results = await conn.execute('select * from system.nodes');
+      await results.getRows();
+      await results.close();
+    } catch (e) {
+      err = e;
+      console.error(e);
+    }
+    await netUp();
+    should.exist(err);
+  }).timeout(ERROR_HANDLING_TEST_TIMEOUT);
+
+  it('14.6 Can detect a network timeout on an open connection', async () => {
+    const newTE = await startTE(JSON.stringify(postData));
+    let err, results;
+    // wait to do anything until the TE is running.
+    let TEProcData;
+    try {
+      TEProcData = await pollForTERunning(newTE.startId)
+    } catch (e) {
+      should.not.exist(e);
+    }
+
+    // create a connection to this TE
+    const conn = await driver.connect({
+      ...config,
+      LBQuery: `round_robin(start_id(${newTE.startId}))`,
+    });
+
+    const tcpKillCommand = `lsof -i tcp:${TEProcData.port} | awk '/node/ {print $9;}' | sed 's/.*:\\([0-9]\\+\\)->.*/\\1/p' -n | tail -n 1 | xargs tcpkill -i lo -9 port 2>/dev/null`;
```

Comment:
Was this technique used by our test suite, or where

Reply:
I believe I saw a network timeout the same as with test 14.5. The tcp kill command needs traffic on the connection before it can kill it, which is why there is not an immediate disconnect error. 

In the test suite I was exposed to we were using tc, but I dropped that approach due to limitations of packages available on the internal RHEL servers (internal package repository is used). This should be a suitable alternative.